pkg/envoy/rds/response.go

```diff
@@ -56,14 +56,17 @@ func NewResponse(catalog catalog.MeshCataloger, proxy *envoy.Proxy, _ *xds_disco
 			return nil, err
 		}
 
-		// All routes from a given source to destination are part of 1 traffic policy between the source and destination.
-		for _, httpRoute := range trafficPolicy.HTTPRoutes {
-			if isSourceService {
-				aggregateRoutesByHost(outboundAggregatedRoutesByHostnames, httpRoute, weightedCluster, hostnames)
-			}
+		// if the service is referenced in a traffic split there will be more than one hostname (root service and backend)
+		for _, hostname := range hostnames {
```

Comment:
The API `GetHostnamesForService` was never meant to return something like: 
`[]string{"foo,bar", "baz,bar"}`

We should fix this API and come up with a cleaner fix. Refer to https://github.com/openservicemesh/osm/issues/1316 which tracks this enhancement. This change doesn't align with #1316 because the API is now returning multiple hostname encodings as a list of strings.

I also see that the same route is applied to all each item in the list returned by `GetHostnamesForService`, so why not just return a list where each element does not have a comma separated list of hostnames?